proto/cosmos/app/v1alpha1/update.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package cosmos.app.v1alpha1;
+
+import "cosmos/app/v1alpha1/config.proto";
+
+// ConfigUpdatePlan describes a planned update to the app config.
+message ConfigUpdatePlan {
+  // height is the block height at which the config update will be performed.
+  int64 height = 1;
+
+  // update are the config updates that will be applied.
+  ConfigUpdates updates = 2;
+}
+
+// ConfigUpdates describes updates to an app config.
+message ConfigUpdates {
+
+  // add_or_update_modules list module configs that will be added if they
+  // describe a new module or updated configs for existing modules with the
+  // same name.
+  repeated ModuleConfig add_or_update_modules = 1;
+
+  // remove_modules is an optional list of app modules to remove.
+  repeated string remove_modules = 2;
```

Comment:
Related to above comment: if order matters, then I think these fields are not enough, it would be hard to e.g. add a new module at 1st position.

TBH we could just simply put the new config as a whole in MsgPlanUpdates. It would be the proposal's job to show diffs.